raven/utilities/forecasting.py

```diff
@@ -173,3 +182,98 @@ def perform_climatology_esp(model_name, forecast_date, forecast_duration, **kwds
     qsims["member"] = (["member"], avail_years)
 
     return qsims
+
+
+def get_recent_ECCC_forecast(region, climate_model='GEPS'):
+    '''
+    This function generates a forecast dataset that can be used to run raven. 
+    Data comes from the ECCC datamart and collected daily. It is aggregated 
+    such that each file contains forecast data for a single day, but for all 
+    forecast timesteps and all members.
+    
+    The code takes the region shapefile and the climate_model to use, here GEPS
+    by default, but eventually could be GEPS, GDPS, REPS or RDPS.
+    '''
+   
+    # Get the file locations and filenames as a function of the climate model and date
+    if climate_model is 'GEPS':
+        
+        # Forecasts will always be recent so start with today!
+        date_now=dt.datetime.now()
+
+        # Find the most recent data on the server by looping from today and go
+        #backwards 5 days. If no data is present, not much value in forecasting.
+        filein=None
+        for i in range(0,7):
+            
+            # Write the string
+            date=date_now-dt.timedelta(days=i)
+            datefcst=dt.datetime.strftime(date,"%Y%m%d")
+            
```

Comment:
I suggest we replace this by a `latest` NcML link as the default.

Reply:
@tlogan2000 It is my understanding that you will be in charge of creating this view so please ping me when it's ready so that I can complete the required changes for this PR.

Reply:
@cjauvin here you can see my limited programming proficiency, as announced... sorry for the extra work and thanks for the support!

Reply:
> @tlogan2000 It is my understanding that you will be in charge of creating this view so please ping me when it's ready so that I can complete the required changes for this PR.

sounds good

Reply:
@richardarsenault It's a pleasure to help, and I'm currently thinking that I really have much to learn, in hydrology and probably a bunch of other stuff!

Reply:
@huard @richardarsenault I have completed my refactoring, but the exact link that needs to be used for the latest view is still in limbo. There's also the question of what dataset we want to use for testing, as using something transient for testing is not ideal, but OTOH having a 2.5 GB file in the testing data is not great either.

Reply:
For my local testing I have downloaded a nc file (as you can see in the code, that needs to be changed) but it's not ideal because I then need to mess with the time series selection dates, etc.

Reply:
I think it should be here: https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/dodsC/birdhouse/testdata/geps_forecast/GEPS_latest.ncml

It seems to have been updated recently so I would guess this is the current location for the file? Unless you need something else. I can provide a much smaller file (by subsetting lat/lon) that could be added to testdata and that would be fixed in time, if that's what you mean?

Reply:
This link is from testdata so I don't think it will be the final, official one.

About your smaller file proposition: yes I think it would be great if it's possible! One thing that we would need to be sure of though is the fact that whatever test we are doing with the smaller dataset offers the same guarantees of correctness than we would have running on the real, bigger dataset.

Also: I'm not sure how `times` should be computed:

https://github.com/Ouranosinc/raven/blob/wps_geomet/raven/utilities/forecasting.py#L244-L246

What should be the starting point?

Reply:
OK, I can make a smaller file, but it would be static (i.e. for a fixed date, so the tests would have to check that file in particular rather than a "live" version. But I think this would be OK if we pair it with a second test that checks to see if the real "latest" file is actually present, contains the correct variables and times. What do you think? The latter would not need to be included in the testdata DB.

Also, for the times: the thing is that the weather forecasts are not uniform. at first they are hourly, then 3-hours and towards the end they are 6-hourly. This is because a climate model is able to be relatively accurate on the hourly scale for 2-3 days, after that it loses in accuracy so going to 3-hours offers the same level of accuracy but with fewer data. Same reasoning for the jump from 3 to 6 hours. But RAVEN does not accept varying time steps, so I simply sample the timeseries at every 6 hours from the starting point. That is what I am doing in the "times" vector. Simply start at 0 (0 hours since the beginning of the forecast) to hour 384 (i.e. 16 days ahead for GEPS) with a step of 6 hours. It's probably not the best option but it works...

Eventually, during maintenance, I'd like to make it so RAVEN is run on the hourly period only (2-3 days), then the final states are saved (the rvc), and then RAVEN is re-run from those states but with the 3-hour data, and so on. So we could maximize the accuracy this way. But that opens another can of worms, regarding hydrograph analysis, graphing, etc.